doc/productivity.md

```diff
@@ -32,6 +33,17 @@ Install `ccache` through your distribution's package manager, and run `./configu
 
 To use ccache for all your C/C++ projects, follow the symlinks method [here](https://ccache.samba.org/manual/latest.html#_run_modes) to set it up.
 
+To get the most out of ccache, put something like this in `~/.ccache/ccache.conf`:
+
+```
+max_size = 50.0G  # or whatever cache size you prefer; default is 5G; 0 means unlimited
+base_dir = /home/yourname  # or wherever you keep your source files
+```
+
+Note: base_dir is required for ccache to share cached compiles of the same file across different repositories / paths; it will only do this for paths under base_dir. So this option is required for effective use of ccache with git worktrees (described below.)
```

Comment:
nit, here and below should be `).`, not `.)`?

Reply:
Even though the way I did it is clearly grammatically not sensible, for some reason I thought it was normal, so I polled some of my friends on IRC and it turns out I'm the only person I know who does it this way. Personal quirk, I guess? Will fix.

Reply:
There are more cases below.

Reply:
Heh, fixed two more instances.